src/renderers/WebGLRenderer.js

```diff
@@ -1853,15 +1854,61 @@ function WebGLRenderer( parameters = {} ) {
 
 	};
 
-	this.setRenderTarget = function ( renderTarget, activeCubeFace = 0, activeMipmapLevel = 0 ) {
+	this.setRenderTarget = function ( renderTarget, activeCubeFace = 0, activeMipmapLevel = 0, options = {} ) {
 
 		_currentRenderTarget = renderTarget;
 		_currentActiveCubeFace = activeCubeFace;
 		_currentActiveMipmapLevel = activeMipmapLevel;
+		const useDefaultFramebuffer = options.framebuffer === undefined;
 
-		if ( renderTarget && properties.get( renderTarget ).__webglFramebuffer === undefined ) {
+		if ( renderTarget ) {
+
+			const renderTargetProperties = properties.get( renderTarget );
+			let hasNewExternalTextures = false;
+
+			if ( options.colorTexture !== undefined ) {
+
+				hasNewExternalTextures = true;
+				properties.get( renderTarget.texture ).__webglTexture = options.colorTexture;
+
+				renderTarget.autoAllocateDepthBuffer = options.depthTexture === undefined;
```

Comment:
It seems `autoAllocateDepthBuffer` can be stored in the render target's properties object. If not, we should at least introduce the property in the class and not add it inline.

Reply:
Done